src/main/java/com/minecraftabnormals/abnormals_core/core/util/BlockUtil.java

```diff
@@ -118,4 +121,42 @@ public static BBRotation getRotationForDirection(Direction currentDirection, Dir
 			}
 		}
 	}
+
+	/**
+	 * <p>Gets the {@link BlockPos} in front of a dispenser.</p>
+	 * {@link BlockState#getBlock() source.getBlockState().getBlock()} must return an instance of {@link DispenserBlock}.
```

Comment:
This is incorrect, the state must have the `DirectionalBlock.FACING` tag.

Reply:
Updated in latest commit